### PR TITLE
Building a custom user reviews page with stars rating in react, typescript and Bootstrap library

### LINKED ISSUE #6767: Building a custom user reviews page with stars rating in react, typescript and Bootstrap library
### Topic Suggestion
Building a custom user reviews page with stars rating in react, typescript and Bootstrap library

#### Proposed title of article
Building a custom user reviews page with stars rating in react, typescript and Bootstrap library

Use keyword research to improve your article's SEO.

#### Proposed article introduction
In front-end web development, most application is expected to be users (client) centric, meaning the overall users reviews and recommendations must be taken into consideration. Most applications rely on user feedback to improve their service quality delivery.
React, typescript and Bootstrap can be an incredible combination in building a custom user review page that users’ recommendations and stars rating can be displayed, similar to apple store and google play store reviews section.
In this article, we will take a deep dive into the various steps and package dependencies required to build a custom user reviews page in react and typescript.

#### Key takeaways 
After reading the article, the reader is expected to understand and be able to implement the below:
-	Getting started with React application development
-	How to add and use typescript in React
-	Adding and using the Bootstrap library dependency in a React application.
-	Building a custom user review page with star ratings in React.js, typescript and Bootstrap
-	How to populate star rating in typescript

#### Article quality
This tutorial is a unique project-based application of React.js and Bootstrap in creating, customizing and styling a web application. This article will feature the building a of multiple components in React.js and how to populate star ratings. 

#### References
Please list links to any published content/research that you intend to use to support/guide this article.

### PR DESCRIPTION
**Pull request by TonyRichardd**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6767

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Please rename your full-page image to use a lowercase file extension.

**Comment by hectorkambow** [comment]
PS: 
@M0nicah  

If you find yourself making any edit to this PR - lets go ahead and close it 👍

File: content/articles/building-a-custom-user-reviews-page-in-react-typescript-and-reactstrap/index.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+
+- Getting started with React application development
+- How to add and use typescript in React
+- Adding and using the reactstrap library in a React application.
+- Building a custom user review page with star ratings in React.js, typescript, and Bootstrap
+- How to populate star rating in typescript
+
+### Prerequisite
+
+Basic knowledge of React, typescript, and a design library is required to follow along with this tutorial. I will also strongly recommend using visual studio code as your preferred text editor. If you do not have the above mentioned here is a link to enroll in a [React crash course]() for free.
```
Comment (hectorkambow) [inline content/articles/building-a-custom-user-reviews-page-in-react-typescript-and-reactstrap/index.md:21]: missing link

**Comment by TonyRichardd** [comment]
@M0nicah I've waited patiently for your reviews. It's been 27 days.
@hectorkambow i have fixed the missing link
Best regards

**Comment by Kayere** [comment]
Congratulations on publishing your first article @TonyRichardd.
Thank you for the review @M0nicah.

**Pull request merged**